src/lib/videotagging_extensions/detection.js

```diff
@@ -0,0 +1,203 @@
+    this.mapVideo = function(frameHandler, exportUntil) {
+        return new Promise((resolve, reject) => {
+            //init canvas buffer
+            var frameCanvas = document.createElement("canvas");
+            frameCanvas.width = self.videotagging.video.videoWidth;
+            frameCanvas.height = self.videotagging.video.videoHeight;
+            var canvasContext = frameCanvas.getContext("2d");
+
+            // start exporting frames using the canplay eventListener
+            self.videotagging.video.oncanplay = iterateFrames;
+            self.videotagging.video.currentTime = 0;
+            self.videotagging.playingCallback();
+
+            //resolve export until
+            var isLastFrame;
+            if (exportUntil === "tagged") {   
+                isLastFrame = function(frameId) {
+                    return (!Object.keys(self.videotagging.frames).length) || (frameId >= parseInt(Object.keys(self.videotagging.frames)[Object.keys(self.videotagging.frames).length-1]));
+                } 
+            }
+            else if (exportUntil === "visited") {
+                isLastFrame = function(frameId) {
+                    var lastVisitedFrameId = Math.max.apply(Math, Array.from(self.visitedFrames));
+                    return (frameId >= lastVisitedFrameId);        
+                }
+            }
+            else { //last
+                isLastFrame = function(frameId) {
+                    return (self.videotagging.video.currentTime >= self.videotagging.video.duration);
+                }
+            }
+            
+
+            function iterateFrames() {
+                var frameId = self.videotagging.getCurrentFrame();
+                var lastFrame = isLastFrame(frameId); 
+                
+                if (lastFrame) {
+                    self.videotagging.video.oncanplay = null;
+                    resolve();
+                }
+                
+                frameHandler(frameId, frameCanvas, canvasContext, () => {
+                    if (!lastFrame) {
+                        self.videotagging.stepFwdClicked(false);
+                    }
+                })
+            }
+        });
+    }
+
+    // TODO: Abstract to a module that receives a "framesReader" object as input
+    //exports frames to the selected detection algorithm format  for model training
+    this.export = function(method, exportUntil, exportPath, testSplit, cb) {
+
+        self.detectionAlgorithmManager.setExporter(method, exportPath, self.videotagging.inputtagsarray, 
+                                            Object.keys(self.videotagging.frames).length,
+                                            self.videotagging.video.videoWidth,
+                                            self.videotagging.video.videoHeight,
+                                            testSplit);
+
+        self.detectionAlgorithmManager.exporter.init()
```

Comment:
Can you bundle setExporter and init together? e.g. change it to setAndInitExporter, or just .initExporter